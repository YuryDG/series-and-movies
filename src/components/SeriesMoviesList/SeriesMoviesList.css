div#infinite-scroll-container > div {  
   display: flex;
   flex-direction: row;
   flex-wrap: wrap;
   justify-content: space-between;
}

.text-center {
   text-align: center;
}