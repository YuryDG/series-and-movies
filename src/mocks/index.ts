export * from './mockApiResponse';
export * from './mockSeriesMovieData';