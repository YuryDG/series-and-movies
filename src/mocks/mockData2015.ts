import { Movie, Serie } from "../types";

export const mockData2015: (Movie | Serie)[] = [
  {
    title: "No Activity",
    description: "No Activity Description",
    programType: "series",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/106/36/10636-PosterArt-9add943c5e62c2d89f6d31458d33508a.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2015
  },
  {
    title: "Better Call Saul",
    description: "Better Call Saul Description",
    programType: "movie",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/175/7/1757-PosterArt-fc0e2a5b18d4a662717f3cbb1827871d.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2015
  },
]