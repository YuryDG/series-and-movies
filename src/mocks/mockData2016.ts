import { Movie, Serie } from "../types";

export const mockData2016: (Movie | Serie)[] = [
  {
    title: "Wolf Creek",
    description: "Wolf Creek Description",
    programType: "series",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2016
  },
  {
    title: "Billions",
    description: "Billions Description",
    programType: "movie",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/117/25/11725-PosterArt-deecf8dbd786dfa2d964413b0bf83726.jpg",
      width: 720,
      height: 1080
    },
    releaseYear: 2016
  },
]