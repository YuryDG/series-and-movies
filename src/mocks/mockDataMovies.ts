import { Movie, Serie } from "../types";

export const mockDataMovies: (Movie | Serie)[] = [
  {
    title: "Billions",
    description: "Billions Description",
    programType: "movie",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/117/25/11725-PosterArt-deecf8dbd786dfa2d964413b0bf83726.jpg",
      width: 720,
      height: 1080
    },
    releaseYear: 2016
  },
  {
    title: "Better Call Saul",
    description: "Better Call Saul Description",
    programType: "movie",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/175/7/1757-PosterArt-fc0e2a5b18d4a662717f3cbb1827871d.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2015
  },
]