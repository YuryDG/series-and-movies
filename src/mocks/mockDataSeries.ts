import { Movie, Serie } from "../types";

export const mockDataSeries: (Movie | Serie)[] = [
  {
    title: "Wolf Creek",
    description: "Wolf Creek Description",
    programType: "series",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2016
  },
  {
    title: "No Activity",
    description: "No Activity Description",
    programType: "series",
    image: {
      url: "https://streamcoimg-a.akamaihd.net/000/106/36/10636-PosterArt-9add943c5e62c2d89f6d31458d33508a.jpg",
      width: 1000,
      height: 1500
    },
    releaseYear: 2015
  }
]