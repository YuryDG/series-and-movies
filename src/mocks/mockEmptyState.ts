import { RootState } from "../app/store";

export const mockEmptyState: RootState = {
  filters: {
    year: 0,
    type: '',
    searchTerm: '',
    sortBy: '',
    pagination: {
      currentPage: 1,
      itemsPerPage: 6
    }
  },
  seriesAndMovies: {
    data: [],
    status: '',
    error: '',
  },
  counter: {
    value: 1,
    status: 'idle'
  }
}