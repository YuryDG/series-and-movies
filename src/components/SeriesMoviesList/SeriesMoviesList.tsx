import InfiniteScroll from "react-infinite-scroller";
import { Movie, Pagination, Serie, SeriesAndMoviesState } from "../../types";
import { SeriesMoviesItem } from "../SeriesMoviesItem";
import './SeriesMoviesList.css';

type SeriesMoviesListProps = {
  data: (Movie | Serie)[],
  pagination: Pagination,
  hasMoreData: boolean;
  dataStatus: SeriesAndMoviesState['status']
  loadMore: (page: number) => void
}

export const SeriesMoviesList: React.FC<SeriesMoviesListProps> = (props) => {
  const { data, hasMoreData, dataStatus, pagination, loadMore } = props;

  if (dataStatus === 'LOADING') {
    return <div className="text-center">
      Loading...
    </div>
  }

  if (dataStatus === 'ERROR') {
    return <div className="text-center">
      Something went wrong.
    </div>
  }

  if (!data.length && dataStatus === 'LOADED') {
    return (
      <div className="text-center">
        <p> No data were found</p>
      </div>
    )
  }

  return (
    <div id="infinite-scroll-container">
      <InfiniteScroll
        pageStart={pagination.currentPage}
        loadMore={loadMore}
        hasMore={hasMoreData}
        loader={<div className="loader" key={0}>Loading ...</div>}
        useWindow={true}
      >
        {data.map((item) => <SeriesMoviesItem key={item.title} item={item} />)}
      </InfiniteScroll>
    </div>
  )
}