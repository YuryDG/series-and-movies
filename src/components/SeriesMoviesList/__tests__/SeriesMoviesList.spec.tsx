import { render } from '@testing-library/react';
import { SeriesMoviesList } from '../SeriesMoviesList';
import { mockDataSeries, mockEmptyState } from '../../../mocks';

describe('SeriesMoviesList', () => {
  it('Should be loading', () => {
    const { container } = render(
      <SeriesMoviesList
        data={[]}
        dataStatus="LOADING"
        pagination={mockEmptyState.filters.pagination}
        hasMoreData={false}
        loadMore={jest.fn()}
      />)
    expect(container).toMatchSnapshot()
  })

  it('should show no data message', () => {
    const { container } = render(
      <SeriesMoviesList
        data={[]}
        dataStatus="LOADED"
        pagination={mockEmptyState.filters.pagination}
        hasMoreData={false}
        loadMore={jest.fn()}
      />)
    expect(container).toMatchSnapshot()
  })

  it('should show error message when error', () => {
    const { container } = render(
      <SeriesMoviesList
        data={[]}
        dataStatus="ERROR"
        pagination={mockEmptyState.filters.pagination}
        hasMoreData={false}
        loadMore={jest.fn()}
      />)
    expect(container).toMatchSnapshot()
  })

  it('should show data when everything is ok', () => {
    const { container } = render(
      <SeriesMoviesList
        data={mockDataSeries}
        dataStatus="LOADED"
        pagination={mockEmptyState.filters.pagination}
        hasMoreData={false}
        loadMore={jest.fn()}
      />)
    expect(container).toMatchSnapshot()
  })
})