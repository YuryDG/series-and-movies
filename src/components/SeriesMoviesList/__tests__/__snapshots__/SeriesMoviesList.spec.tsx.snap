// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesMoviesList Should be loading 1`] = `
<div>
  <div
    class="text-center"
  >
    Loading...
  </div>
</div>
`;

exports[`SeriesMoviesList should show data when everything is ok 1`] = `
<div>
  <div
    id="infinite-scroll-container"
  >
    <div>
      <div
        class="item-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="Wolf Creek"
            height="250"
            src="https://streamcoimg-a.akamaihd.net/000/128/61/12861-PosterArt-ec32a81986a45eac7e080112075ab466.jpg"
          />
        </div>
        <div
          class="content"
        >
          <p
            class="info"
          >
            <span
              class="title"
            >
              Wolf Creek
            </span>
            <span
              class="type"
            >
              Serie
               - 
              2016
            </span>
          </p>
          <p>
            Wolf Creek Description
          </p>
        </div>
      </div>
      <div
        class="item-container"
      >
        <div
          class="image-container"
        >
          <img
            alt="No Activity"
            height="250"
            src="https://streamcoimg-a.akamaihd.net/000/106/36/10636-PosterArt-9add943c5e62c2d89f6d31458d33508a.jpg"
          />
        </div>
        <div
          class="content"
        >
          <p
            class="info"
          >
            <span
              class="title"
            >
              No Activity
            </span>
            <span
              class="type"
            >
              Serie
               - 
              2015
            </span>
          </p>
          <p>
            No Activity Description
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SeriesMoviesList should show error message when error 1`] = `
<div>
  <div
    class="text-center"
  >
    Something went wrong.
  </div>
</div>
`;

exports[`SeriesMoviesList should show no data message 1`] = `
<div>
  <div
    class="text-center"
  >
    <p>
       No data were found
    </p>
  </div>
</div>
`;
